### PR TITLE
Add SecretValue, update for latest ndc-test

### PR DESCRIPTION
**Pull request by paf31**
This ought to have been two PRs, but these will go out together in the nest `ndc-spec` tag anyway.

- Update for latest `ndc-test` - add `snapshot-dir` argument and new `replay` command to `default_main`
- Add `SecretValue` type here instead of in `ndc-spec`

### CODE REVIEW

File: rust-connector-sdk/src/secret.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        if let Schema::Object(o) = &mut s {
+            if let Some(m) = &mut o.metadata {
+                m.id = Some(Self::schema_id().into());
+            }
+        }
+        s
+    }
+
+    fn schema_id() -> std::borrow::Cow<'static, str> {
+        "hasura.io/SecretValue".into()
```
Comment (abhinav-hasura) [inline rust-connector-sdk/src/secret.rs:38]: Looks like this needs to be a proper URI reference (but not necessary network reachable) according to the JSONSchema spec.

How about: "https://hasura.io/jsonschemas/SecretValue"?

**Review by abhinav-hasura — APPROVED** [review]

File: rust-connector-sdk/src/secret.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+use schemars::{self, schema::Schema, JsonSchema};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+#[schemars(title = "SecretValue")]
+/// Either a literal string or a reference to a Hasura secret
+pub enum SecretValueImpl {
+    Value(String),
+    StringValueFromSecret(String),
```
Comment (SamirTalwar) [inline rust-connector-sdk/src/secret.rs:10]: Will this ever be constructed, or is this purely for generating the JSON schema?

Comment (abhinav-hasura) [reply rust-connector-sdk/src/secret.rs:10]: Right now I don't think anyone constructs this, but at some point I can see MBS constructing this when deserializing a secretable value.

**Comment by paf31** [comment]
Will split this from the `ndc-test` change and make a separate PR.

**Pull request closed without merging**